File: presentation/src/main/java/com/popularmovies/vpaliy/popularmoviesapp/ui/view/RatioImageView.java
package com.popularmovies.vpaliy.popularmoviesapp.ui.view;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.v7.widget.AppCompatImageView;
import android.util.AttributeSet;

import com.popularmovies.vpaliy.popularmoviesapp.R;

public class RatioImageView extends AppCompatImageView {

    //16:9
    private float ratio =.5625f;

    public RatioImageView (Context context){
        this(context,null);
    }

    public RatioImageView (Context context, AttributeSet attrs){
        this(context,attrs,0);
    }

    public RatioImageView (Context context, AttributeSet attrs, int defStyle){
        super(context,attrs,defStyle);
        if(attrs!=null){
            TypedArray array=context.obtainStyledAttributes(attrs, R.styleable.RatioImageView);
            this.ratio=array.getFloat(R.styleable.RatioImageView_image_ratio,ratio);
            array.recycle();
        }
    }

    public void setRatio(float ratio) {
        this.ratio = ratio;
        requestLayout();
    }
    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        int measuredWidth=getMeasuredWidth();
        setMeasuredDimension(measuredWidth, Math.round(measuredWidth*ratio));
    }
}

File: presentation/src/main/java/com/popularmovies/vpaliy/popularmoviesapp/ui/view/RatioImageView.kt
package com.popularmovies.vpaliy.popularmoviesapp.ui.view

import android.content.Context
import android.support.v7.widget.AppCompatImageView
import android.util.AttributeSet

import com.popularmovies.vpaliy.popularmoviesapp.R

class RatioImageView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyle: Int = 0):
        AppCompatImageView(context, attrs, defStyle) {

    //16:9
    private var ratio = .5625f

    init {
        if (attrs != null) {
            val array = context.obtainStyledAttributes(attrs, R.styleable.RatioImageView)
            this.ratio = array.getFloat(R.styleable.RatioImageView_image_ratio, ratio)
            array.recycle()
        }
    }

    fun setRatio(ratio: Float) {
        this.ratio = ratio
        requestLayout()
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val measuredWidth = measuredWidth
        setMeasuredDimension(measuredWidth, Math.round(measuredWidth * ratio))
    }
}